packages/cli/src/commands/build.ts

```diff
@@ -67,43 +70,76 @@ export const build: Command = {
   },
 };
 
+async function validateManifestModules(polywrapManifest: PolywrapManifest) {
+  if (
+    polywrapManifest.project.type !== "interface" &&
+    !polywrapManifest.source.module
+  ) {
+    const missingModuleMessage = intlMsg.lib_compiler_missingModule();
+    throw Error(missingModuleMessage);
+  }
+
+  if (
+    polywrapManifest.project.type === "interface" &&
+    polywrapManifest.source.module
+  ) {
+    const noInterfaceModule = intlMsg.lib_compiler_noInterfaceModule();
+    throw Error(noInterfaceModule);
+  }
+}
+
 async function run(options: BuildCommandOptions) {
-  const { watch, verbose, manifestFile, outputDir, clientConfig } = options;
+  const {
+    watch,
+    verbose,
+    manifestFile,
+    outputDir,
+    clientConfig,
+    docker,
+  } = options;
 
   // Get Client
   const client = new PolywrapClient(clientConfig);
 
-  // Ensure docker is installed
-  if (!isDockerInstalled()) {
-    console.log(intlMsg.lib_docker_noInstall());
-    return;
-  }
-
   const project = new PolywrapProject({
     rootDir: path.dirname(manifestFile),
     polywrapManifestPath: manifestFile,
     quiet: !verbose,
   });
   await project.validate();
 
-  const dockerLock = new FileLock(
-    project.getCachePath("build/DOCKER_LOCK"),
-    print.error
-  );
+  const polywrapManifest = await project.getManifest();
+  await validateManifestModules(polywrapManifest);
 
-  const schemaComposer = new SchemaComposer({
+  const buildStrategy = createSourceBuildStrategy(docker ? "docker" : "local", {
```

Comment:
It might even be clearer if there was no `createSourceBuildStrategy` func and it's code was just here.
Wdyt?

Reply:
Yep, agreed